src/federation_server.js

```diff
@@ -161,7 +163,9 @@ export class FederationServer {
   }
 
   _sendRequest(url) {
-    return axios.get(url.toString(), {maxContentLength: FEDERATION_RESPONSE_MAX_SIZE})
+    let timeout = this.timeout | Config.getTimeout();
```

Comment:
`this.timeout` is not set anywhere.

In general it should work like this:
1. In `constructor` set `this.timeout` to `opts.timeout` or `Config.getTimeout()` (if former not set).
2. `static` method create `FederationServer` instance internally so we should just pass `opts` to class constructor.
3. Finally `_sendRequest` should just use `this.timeout`. No need to check `Config` again.

Also, should be `||`. `|` is bitwise OR.